# latest_volume_pickups.pkl Data Exploration

In [1]:
import pickle

In [2]:
latest_volume_pickups = pickle.load(open('./Data/features_condition/latest_volume_pickups.pkl', 'rb'))

volume_pickups_keys = list(latest_volume_pickups.keys())
print(f'Number of keys: {len(volume_pickups_keys)}')

print(volume_pickups_keys[:20])

Number of keys: 6602688
[(36, 75, 161, 4), (44, 34, 176, 6), (31, 64, 151, 4), (26, 64, 137, 5), (43, 70, 168, 4), (25, 35, 245, 1), (35, 30, 272, 4), (17, 16, 202, 1), (45, 25, 184, 5), (35, 85, 103, 4), (6, 80, 60, 1), (29, 19, 266, 5), (46, 40, 226, 1), (7, 19, 110, 5), (26, 46, 109, 2), (20, 53, 7, 4), (28, 76, 148, 5), (18, 13, 34, 2), (32, 50, 287, 4), (29, 53, 188, 5)]


# Key index definition
- 0: x_grid index
- 1: y_grid index
- 2: time bucket $\in$ [0,287]
- 3: day-of-week index

# Value index definition

In [4]:
val_element_list = ['Number of pickups', 'Traffic volume']

print(f'For example key {volume_pickups_keys[0]}')
for i, value in enumerate(latest_volume_pickups[volume_pickups_keys[0]]):
    print(f'Index [{0}] = {latest_volume_pickups[volume_pickups_keys[0]][i]} -- {val_element_list[i]}')

For example key (36, 75, 161, 4)
Index [0] = 0 -- Number of pickups
Index [0] = 15 -- Traffic volume
